.github/workflows/ci.yml

```diff
@@ -0,0 +1,121 @@
+    branches:
+      - develop
+      - backports/**
+    paths-ignore:
+      - '**.rst'
+      - '**.md'
+      - 'scripts/dev'
+  merge_group:
+    types: [checks_requested]
+
+concurrency:
+  group: ${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: CI
+    strategy:
+      matrix:
+        image: ['ubuntu-cuda', 'centos-rocm']
+        buildtype: ['debug', 'ndebug']
+        geometry: ['orange', 'vecgeom']
+        special: [null]
+        exclude:
+          - image: 'centos-rocm'
+            geometry: 'vecgeom'
+        include:
+          - image: 'centos-rocm'
+            buildtype: 'asan'
+            geometry: 'orange'
+          - image: 'centos-rocm'
+            buildtype: 'debug'
+            geometry: 'orange'
+            special: 'float'
+          - image: 'centos-rocm'
+            buildtype: 'reldeb'
+            geometry: 'geant4'
+          - image: 'ubuntu-cuda'
+            buildtype: 'reldeb'
+            geometry: 'vecgeom'
+    env:
+      ASAN_OPTIONS: "detect_leaks=0"
+        #      CCACHE_DIR: "${{github.workspace}}/.ccache"
+        #      CCACHE_MAXSIZE: "10G"
+      CELER_TEST_STRICT: 1
+      CELER_DISABLE_DEVICE: 1 # IMPORTANT
+      CMAKE_PRESET: >-
+        ${{matrix.buildtype}}-${{matrix.geometry}}${{matrix.special && '-' || ''}}${{matrix.special}}
+    runs-on: ubuntu-latest
+    container:
+      image: >-
+        docker.io/celeritas/${{
+             matrix.image == 'ubuntu-cuda' && 'ci-jammy-cuda11:2023-08-02'
+          || matrix.image == 'centos-rocm' && 'ci-centos7-rocm5:2022-12-14.2'
+          || null
+        }}
+      # See https://github.com/actions/checkout/issues/956
+      options: --user root
+    steps:
+      - name: Set up environment
+        run: |
+          . /etc/profile
+          echo "/opt/view/bin" >> $GITHUB_PATH
+          echo "CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}" >> $GITHUB_ENV
+      # NOTE: checkout must occur *after* setting up environment for git tags to work
+      # NOTE: checkout v4 fails: https://github.com/actions/checkout/issues/1487
+      # NOTE: depth must be enough to include the previous tag
+      - name: Check out Celeritas
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 255
+          fetch-tags: true
+      - name: Configure Celeritas
+        run: |
+          git config --global --add safe.directory ${PWD}
+          ln -fs scripts/cmake-presets/ci-${{matrix.image}}.json CMakeUserPresets.json
+          cmake --preset=${CMAKE_PRESET}
+      - name: Build Celeritas
+        run: |
+          cmake --build --preset=${CMAKE_PRESET}
+      - name: Test Celeritas
+        working-directory: build
+        run: |
+          ctest --timeout 180 --output-on-failure \
+            --test-output-size-passed=65536 --test-output-size-failed=1048576
+      - name: Install Celeritas
+        run: |
+          . /etc/profile
```

Comment:
Why is this step setting (again?) the environment?

Reply:
`/etc/profile` doesn't get loaded by the github runner because it isn't using the entrypoint for some reason. This is necessary to set up the spack envrionment so we can get `/opt/view/bin` (and whatever else) into the path.

After the smoke clears from the R&D evaluation I'm planning to redo the CI images so this hopefully isn't necessary.

Reply:
The weird looking part is that it is call here but not in other runs (in particular `- name: Build Celeritas` and `Test Celeritas`)

Reply:
Ahhh. OK it might no longer be necessary, but the build/install hooks are using whatever CMake cached from its initial run, rather than depending on the environment. I'll try deleting it.